pkg/services/ngalert/metrics/historian.go

```diff
@@ -0,0 +1,23 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/weaveworks/common/instrument"
+)
+
+type Historian struct {
+	WriteDuration *prometheus.HistogramVec
+}
+
+func NewHistorianMetrics(r prometheus.Registerer) *Historian {
+	return &Historian{
+		WriteDuration: promauto.With(r).NewHistogramVec(prometheus.HistogramOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      "state_history_request_duration_seconds",
```

Comment:
I settled on `request_duration` over `write_duration`/`persist_duration` because this histogram covers ALL requests - both writes and reads. They can be differentiated by the operation label that is provided by weaveworks.

Reply:
The operation label will translate the path unless specified. I think this is fine? 

It'll require the user of the metrics to have some knowledge of how the Loki API works but I think not hiding it behind a "write" or "read" abstraction is actually better. The user will know exactly which endpoints grafana is calling.

Reply:
I noticed this when i tested the feature and decided that I liked it. It also contains the method - it looks like:
`operation="GET /api/some/loki/endpoint"`